### ISSUE TITLE
KNN classifier 

### ISSUE THREAD

**Issue by Hammadwakeel** [Hammadwakeel/Programming-for-Ai#3]
k nearest neighbor classifier python code from scratch

#Importing important Libraries
import numpy as np
import pandas as pd
import sklearn

#Loading the dataset to the jupiter nootbook
dataset = pd.read_csv('Social_Network_Ads.csv')

#Selecting features from the dataset
X = dataset.loc[:, ['Gender','Age' ,'EstimatedSalary' ]].values

#Selecting Labels from the dataset
y = dataset.loc[:,'Purchased'].values

#Converting the gender from String to int i.e from male, female to 0,1
from sklearn.preprocessing import LabelEncoder
le = LabelEncoder()
X[:,0] = le.fit_transform(X[:,0])

#Spliting the data to training and testing
from sklearn.model_selection import train_test_split
X_train, X_test, y_train, y_test = train_test_split(X, y, test_size = 0.20, random_state = 0)

#Scaling the data 
from sklearn.preprocessing import StandardScaler
sc = StandardScaler()
X_train = sc.fit_transform(X_train)
X_test = sc.transform(X_test)

#importing the classifiers and selecting KNeighborsClassifier
from sklearn.neighbors import KNeighborsClassifier
classifier = KNeighborsClassifier(n_neighbors = 5, metric = 'minkowski', p = 2)
classifier.fit(X_train, y_train)

#testing the data 
y_pred = classifier.predict(X_test)
y_pred

#checking the accuracy and creating confusion  matrix
from sklearn.metrics import confusion_matrix,accuracy_score
cm = confusion_matrix(y_test, y_pred)
ac = accuracy_score(y_test,y_pred)
cm

ac

dataset= [Social_Network_Ads.csv](https://github.com/Hammadwakeel/Programming-for-Ai/files/9844961/Social_Network_Ads.csv)
